hs_collection_resource/views.py

```diff
@@ -0,0 +1,103 @@
+            downgrade = False
+
+            if len(collection_content_res_id_list) == 0:
+                downgrade = True
+            else:
+                for checked_res_id in collection_content_res_id_list:
+                    res_checked = hydroshare.get_resource_by_shortkey(checked_res_id)
+                    if not res_checked.raccess.public and not res_checked.raccess.discoverable:
+                        downgrade = True
+                        break
+
+            if downgrade:
+                set_resource_flag(request, collection_res_obj.short_id, t="make_private")
+                new_sharing_status = "Private"
+
+        # check if current user is owner or super_user (owner and super_user can change sharing status on frontend)
+        user_permission = "Edit"
+        if collection_res_obj in user.uaccess.owned_resources or user.is_superuser:
+            user_permission = "Own"
+
+        if collection_res_obj.metadata.has_all_required_elements():
+            metadata_status = "Sufficient to make public"
+        else:
+            metadata_status = "Insufficient to make public"
+
+        resource_modified(collection_res_obj, user)
+        ajax_response_data = {'status': 'success', 'user_permission': user_permission, \
+                              'current_sharing_status': current_sharing_status, \
+                              'new_sharing_status': new_sharing_status, 'metadata_status': metadata_status}
+        return JsonResponse(ajax_response_data)
+
+    except Exception as ex:
+        logger.exception("update_collection: %s" % (ex.message))
+        ajax_response_data = {'status': 'error', 'message': ex.message}
+        return JsonResponse(ajax_response_data)
+
+# loop through member resources in collection ("shortkey") to check if the target user ("user_id") has
+# at least View permission over them.
+def collection_member_permission(request, shortkey, user_id, *args, **kwargs):
+    try:
+        collection_res_obj, is_authorized, user = authorize(request, shortkey,
+                                              needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE,
+                                              raises_exception=True)
+        no_permission_list = []
+
+        user_to_share_with = user_from_id(user_id)
+        if collection_res_obj.metadata.collection:
+            for res_in_collection in collection_res_obj.metadata.collection.resources.all():
+                if not user_to_share_with.uaccess.can_view_resource(res_in_collection) \
+                   and not res_in_collection.raccess.public and not res_in_collection.raccess.discoverable:
+                    no_permission_list.append(res_in_collection.short_id)
+            status = "success"
+            ajax_response_data = {'status': status, 'no_permission_list': no_permission_list}
+            return JsonResponse(ajax_response_data)
+        else:
+            raise Exception("Collection element is not yet initialized.")
```

Comment:
Shouldn't raise exception if this view function is to be used in an ajax call which its seems. Instead ```return JsonResponese(with the exception message)```.  Again this function should have only one ```return``` statement.